salt/modules/solarisips.py

```diff
@@ -366,8 +366,8 @@ def latest_version(*names, **kwargs):
     ret.update(zip(unmatched, [ '' ] * len(unmatched)))
 
     # Return a string if only one package name passed
-    if len(names) == 1:
-        return ret[names[0]]
+    if len(ret) == 1:
```

Comment:
This is wrong.   The number of results does not determine the output.   The number of inputs determines the output type.   One Package name provided return a string, more than one package name return a dict. The len(name) == 1 was correct.